### ISSUE TITLE
missing_order_features

### ISSUE THREAD

**Issue by 8ball030** [8ball030/ethereal_rust_sdk#29]
- [x] post only
- [x] reduce only
- [ ] stop loss / take profit
- [x] expiration time
- [ ] market orders

**Comment by yieldenjoyer** [comment]
# MARKET Take Profit - executes as MARKET when stop_price triggered
  take_profit_market = await client.create_order(
      order_type="MARKET",      # MARKET order when triggered
      product_id=pid,
      side=1,                   # Sell to exit long
      quantity=qty,
      reduce_only=True,         # Only closes position
      stop_type=0,              # 0 = GAIN (take profit)
      stop_price=Decimal("105000")  # Triggers when price >= this
  )

  The difference:
  - order_type="MARKET" = when stop triggers, fills at market (guaranteed fill except for
  slippage & fees)
  - order_type="LIMIT" + price = when stop triggers, becomes limit order (might not fill
  if price moves, then i'd recommend a limit chase to close if your looking to TP while minimizing slippage.

  reduce_only is independent, you can have:
  - order_type="MARKET" + reduce_only=True = Market TP
  - order_type="LIMIT" + reduce_only=True = Limit TP (my preferred)

  For OCO with market TPs:
  # Market TP 
  tp = await client.create_order(
      order_type="MARKET",  # Market fill
      stop_type=0,          # GAIN
      stop_price=Decimal("105000"),
      reduce_only=True,
      group_id=group_id,
      group_contingency_type=1  # OCO
  )

  # Market SL
  sl = await client.create_order(
      order_type="MARKET",  # Market fill
      stop_type=1,          # LOSS
      stop_price=Decimal("95000"),
      reduce_only=True,
      group_id=group_id,
      group_contingency_type=1  # OCO
  )

1. TAKER (Market Order)

  # Immediate fill at best available price
  market_buy = await client.create_order(
      order_type="MARKET",  # Taker - crosses spread
      quantity=Decimal("0.01"),
      side=0,  # 0=buy, 1=sell
      ticker="BTCUSD"
  )

  Rust struct equivalent:
  struct MarketOrder {
      order_type: String,  // "MARKET"
      quantity: Decimal,
      side: u8,  // 0=buy, 1=sell
      ticker: String,
  }

  2. STOP LOSS

  # Triggers when price falls BELOW stop_price (for long position)
  stop_loss = await client.create_order(
      order_type="MARKET",
      product_id=pid,
      side=1,  # Sell to exit long
      quantity=qty,
      reduce_only=True,  # CRITICAL: prevents opening new position
      stop_type=1,  # 1 = LOSS
      stop_price=Decimal("95000")  # Trigger price
  )

  3. TAKE PROFIT
  # Triggers when price rises ABOVE stop_price (for long position)
  take_profit = await client.create_order(
      order_type="MARKET",
      product_id=pid,
      side=1,  # Sell short to exit long (example)
      quantity=qty,
      reduce_only=True,
      stop_type=0,  # 0 = GAIN
      stop_price=Decimal("105000")  # Target price
  )

  4. OCO (One-Cancels-Other) - TP + SL Linked

  import uuid
  group_id = str(uuid.uuid4())  # Links the orders together

  # Take profit (triggers on gain)
  tp = await client.create_order(
      order_type="LIMIT",
      product_id=pid,
      side=1,
      quantity=Decimal("0.001"),
      price=Decimal("105000"),  # Exit price
      time_in_force="GTD",
      reduce_only=True,
      group_id=group_id,
      group_contingency_type=1  # 1 = OCO
  )

  # Stop loss
  sl = await client.create_order(
      order_type="LIMIT",
      product_id=pid,
      side=1,
      quantity=Decimal("0.001"),
      price=Decimal("95000"),  # Exit price
      time_in_force="GTD",
      reduce_only=True,
      group_id=group_id,
      group_contingency_type=1  # 1 = OCO
  )
  # When one fills, the other auto-cancels

  Python Implementation (place_order signature):
  def place_order(self,
                 market: str,
                 side: str,
                 size: Decimal,
                 price: Optional[Decimal] = None,
                 order_type: str = 'limit',
                 post_only: bool = True,
                 reduce_only: bool = False,
                 time_in_force: str = 'GTD',
                 group_id: Optional[str] = None,
                 group_contingency_type: Optional[int] = None,
                 stop_price: Optional[Decimal] = None,  # NEW
                 stop_type: Optional[int] = None,      
                 ...):

  # Pass to SDK:
  if stop_price is not None:
      order_params['stop_price'] = float(stop_price)
  if stop_type is not None:
      order_params['stop_type'] = stop_type
